### DISCUSSION TITLE
Architecture Diagram

### DISCUSSION THREAD

**Discussion by johija** [toluaina/pgsync#355] [General]
Hi @toluaina,
Is there any architecture or flow diagram available to understand better the components used in pgsync?

Thank you!

**Comment by toluaina** [comment]
Afraid I don't have one at present. I am working on it. 
I am however happy to answer any specific questions you have in detail.

**Reply by johija** [reply]
Thanks, can you check these:
1. I think Write Ahead Log is used during updates. But in the initial sync when there are no updates but snapshot read and write does pgsync still use WAL? 
2. What is the use of redis?
3. Planning to run PGsync in a AWS EC2 server. If a new EC2 server comes up will the full indexing happen every time?
